### PR TITLE
jmeter listener enhanced to support running in distributed mode

### PR DESCRIPTION
**Pull request by howardosborne**

### CODE REVIEW

File: locust_plugins/jmeter_listener.py [hunk 1]
```diff
@@ -40,9 +40,7 @@ def __init__(
         self.flush_size = flush_size
         # results filename format
         self.results_timestamp_format = "%Y_%m_%d_%H_%M_%S"
-        self.results_filename = (
-            "results_" + datetime.fromtimestamp(time()).strftime(self.results_timestamp_format) + ".csv"
-        )
+        self.results_filename = (f'results_{datetime.fromtimestamp(time()).strftime(self.results_timestamp_format)}.csv')
```
Comment (cyberw) [inline locust_plugins/jmeter_listener.py:43]: Accidental list here? :)

File: locust_plugins/jmeter_listener.py [hunk 2]
```diff
@@ -99,7 +103,7 @@ def _flush_to_log(self):
         self.results_file.flush()
         self.csv_results = []
 
-    def _write_final_log(self):
+    def _write_final_log(self, **kwargs):
```
Comment (cyberw) [inline locust_plugins/jmeter_listener.py:106]: Why accept kwargs that is not used?

File: locust_plugins/jmeter_listener.py [hunk 3]
```diff
@@ -136,11 +140,20 @@ def add_result(self, success, _request_type, name, response_time, response_lengt
             connect,
         ]
         self.csv_results.append(self.field_delimiter.join(row))
-        if len(self.csv_results) >= self.flush_size:
+        if len(self.csv_results) >= self.flush_size and self.client_type != "WorkerRunner":
```
Comment (cyberw) [inline locust_plugins/jmeter_listener.py:143]: Use typeof instead of string comparison

**Comment by cyberw** [comment]
Right now the build is failing on formatting. please use Black to do automatic formatting, and after that I’m pretty sure Pylint will have some suggestions as well (I should really change the order in which those are run in the build)

But I like the core of the PR!

File: locust_plugins/jmeter_listener.py (hunk 1 above)
Comment (howardosborne) [reply locust_plugins/jmeter_listener.py:43]: oops - corrected

File: locust_plugins/jmeter_listener.py (hunk 2 above)
Comment (howardosborne) [reply locust_plugins/jmeter_listener.py:106]: Thanks - I had an idea and backed it out...

File: locust_plugins/jmeter_listener.py (hunk 3 above)
Comment (howardosborne) [reply locust_plugins/jmeter_listener.py:143]: do you mean isinstance? - If so, I think this needs an explicit import to be able to use it.

Comment (cyberw) [reply locust_plugins/jmeter_listener.py:143]: I meant `type(thing) is WorkerRunner` (not typeof, that is Java :)

isinstance or type() work equally well in this case. Yes, they require an import, but it is still nicer than string compare...

File: locust_plugins/jmeter_listener.py (hunk 2 above)
Comment (howardosborne) [reply locust_plugins/jmeter_listener.py:106]: Looked at it further. This method is using an event hook so needs to accommodate whatever is passed in.

Comment (cyberw) [reply locust_plugins/jmeter_listener.py:110]: Ok! If pylint gives you attitude about unused parameter, you can name it _kwargs instead.

Comment (howardosborne) [reply locust_plugins/jmeter_listener.py:110]: without the unpacking operator?

Comment (cyberw) [reply locust_plugins/jmeter_listener.py:110]: no, with. so **_kwargs
Sorry if I was unclear..

Comment (cyberw) [reply locust_plugins/jmeter_listener.py]: You still have a couple more unused arguments, other than that it is looking good!

File: locust_plugins/jmeter_listener.py [hunk 4]
```diff
@@ -136,11 +144,22 @@ def add_result(self, success, _request_type, name, response_time, response_lengt
...
             self._flush_to_log()
 
     def _request_success(self, request_type, name, response_time, response_length, **kw):
         self.add_result("true", request_type, name, response_time, response_length, "", **kw)
 
     def _request_failure(self, request_type, name, response_time, response_length, exception, **kw):
         self.add_result("false", request_type, name, response_time, response_length, str(exception), **kw)
+
+    def _report_to_master(self, client_id, data):
+        client_id = client_id
```
Comment (cyberw) [inline locust_plugins/jmeter_listener.py:157]: This is weird :)

Comment (howardosborne) [reply locust_plugins/jmeter_listener.py:157]: happy to take suggestions...

Comment (cyberw) [reply locust_plugins/jmeter_listener.py:157]: I just meant: why are you assigning a variable to itself?

Comment (howardosborne) [reply locust_plugins/jmeter_listener.py:157]: ...which is how I had it to start with. I'd appreciate some guidance on how this method should look. Thanks

Comment (cyberw) [reply locust_plugins/jmeter_listener.py:157]: Just name the variable _client_id if you all you want is to indicate that it is unused (and remove the client_id = client_id line)

Comment (howardosborne) [reply locust_plugins/jmeter_listener.py]: I see it's a keyword argument so I just added it to _kwargs

**Comment by cyberw** [comment]
Thanks!

**Pull request merged**